GitHub actions port 2

chick opened the pull request:
**Related issue**: <!-- if applicable -->

<!-- choose one -->
**Type of change**: bug fix | new feature | other enhancement

<!-- choose one -->
**Impact**: rtl change | software change | unknown | other

**Release Notes**
<!-- Text from here to the end of the body will be considered for inclusion in the release notes for the version containing this pull request. -->

darius-bluespec committed:
Support specifying architecture when building tools.

Add --arch option to build-toolchains.sh to support building
other toolchain variants (e.g., rv32gc).

davidbiancolin committed:
Update frequency selection related comments

abejgonzalez committed:
Use "tile" instead of "core" to assign freq's

jerryz123 committed:
Bump gemmini for config changes

jerryz123 committed:
Bump esp-isa-sim

abejgonzalez committed:
Small comments to Clocks.scala

abejgonzalez committed:
First attempt at getting Offchip AXI port

abejgonzalez committed:
Enlarge serial width | Bugfix loadmem disable | Add TracerV

abejgonzalez committed:
General renaming / cleanup

abejgonzalez committed:
Fix FireChip compilation | Remove extra DefaultSerialTL in bridges

abejgonzalez committed:
Cleanup | Fix BlockDevice clocking issues

abejgonzalez committed:
Merge pull request #807 from ucb-bar/tile_clk_assignment_fix

Use "tile" instead of "core" to assign freq's in WithTileFrequency config. fragment

abejgonzalez committed:
Slightly cleaner implementation

abejgonzalez committed:
Add support for harness pll

abejgonzalez committed:
1st pass at connecting to harness PLL | Put UART adapter on harnessClock/Reset

abejgonzalez committed:
Cleanup config + fragments | Remove reference clk div/rst catch in harness [ci skip]

abejgonzalez committed:
Distinguish between implicit clock/reset and reference harnessClock/Reset | Don't use parameter system to pass referenceFreq

jerryz123 committed:
Merge pull request #808 from ucb-bar/gemmini-bump

Bump gemmini for config changes

abejgonzalez committed:
Update FireSim to support harness clocks | Small config renaming

abejgonzalez committed:
Merge pull request #805 from ucb-bar/davidbiancolin-patch-1

[FireChip] Update frequency selection related comments

abejgonzalez committed:
Renaming updates | Have FireSim clocks request frequency by default

abejgonzalez committed:
First doc pass (no updated imgs) [ci skip]

abejgonzalez committed:
Update doc images [ci skip]

abejgonzalez committed:
Merge remote-tracking branch 'origin/dev' into offchip-axi-setup

abejgonzalez committed:
Fix docs harness binders reference

jerryz123 committed:
Add fragment to configure PMPs

Surprisingly there is no existing fragment to do this defined in rocketchip. Add our own here.

abejgonzalez committed:
Clean up the chip communication docs a bit more [ci skip]

jerryz123 committed:
Move TileResetCtrl before the ResetSynchronizers, and give them an async reset

KPrabs106 committed:
Fix IOCell generation for clock and reset to use IOCellKey

jerryz123 committed:
Merge pull request #824 from KPrabs106/dev

Fix IOCell generation for clock and reset to use IOCellKey

chick committed:
Port CircleCI to Github actions
- Make copies of all script files
  - These maybe unifiable later
- Working on script chipyard-ci-process.yml

abejgonzalez committed:
Merge pull request #821 from ucb-bar/pmp-fragment

Add fragment to configure PMPs

abejgonzalez committed:
Allow the PLL to request the max freq

abejgonzalez committed:
More clarifications on harness clocks

abejgonzalez committed:
Merge pull request #802 from darius-bluespec/build-toolchains-arch

Support specifying architecture when building tools.

chick committed:
Port

chick committed:
keys error

jerryz123 committed:
Bump testchipip for TileResetCtrl changes

jerryz123 committed:
Bump testchipip

chick committed:
keys error

chick committed:
keys error

chick committed:
try fix perms up front

chick committed:
get more setup info

chick committed:
try workflow instead of workspace

chick committed:
trying somewhere else

jerryz123 committed:
Merge pull request #826 from ucb-bar/tile-reset-async

Fix TileResetCtrl to be ahead of reset synchronizers.

jerryz123 committed:
Add option to add async queues between chip-serialIO and harness serdes

jerryz123 committed:
Update BridgeBinders with new HarnessRAM clocking

jerryz123 committed:
Bump testchipip

jerryz123 committed:
Fix SerialTL HarnessRAM BridgeBinder

jerryz123 committed:
Give HarnessRAM implicit clock/reset in SerialTiedOff

abejgonzalez committed:
Small renaming/cleanup | Use LinkedHashMaps

abejgonzalez committed:
Small renaming in docs

abejgonzalez committed:
Use def instead of var Option for ref frequency

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
play with cacheing toolchain

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

abejgonzalez committed:
Merge remote-tracking branch 'origin/dev' into offchip-axi-setup

abejgonzalez committed:
Small spacing fixes

abejgonzalez committed:
Update docs/Advanced-Concepts/Chip-Communication.rst

Co-authored-by: alonamid <alonamid@eecs.berkeley.edu>

abejgonzalez committed:
Splitting up FireSim default frequencies into a separate config frag.

abejgonzalez committed:
Allow run-asm/bmark debug make targets to specify random seed

abejgonzalez committed:
Fix TileResetCtrl so that tiles come out of reset after rest of uncore

abejgonzalez committed:
Merge branch 'offchip-axi-setup' of github.com:ucb-bar/chipyard into offchip-axi-setup

abejgonzalez committed:
Bump testchipip

abejgonzalez committed:
Merge pull request #828 from ucb-bar/async-serial

Add option to add async queues between chip-serialIO and harness serdes

abejgonzalez committed:
Use async queue to connect serdesser + other components

abejgonzalez committed:
Bump testchipip

abejgonzalez committed:
Update harnessClk/Rst naming to buildtop | Small docs cleanup

abejgonzalez committed:
Merge pull request #812 from ucb-bar/offchip-axi-setup

AXI Memory Over TL Serial Link

jerryz123 committed:
Don't pass JTAG oe signal off-chip (#832)

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
testing

chick committed:
try set-output

chick committed:
try set-output

chick committed:
try set-output

chick committed:
try set-output

chick committed:
try set-output

chick committed:
try set-output

chick committed:
force-build

chick committed:
$HOME BROKEN

chick committed:
build riscv-tools 1

dependabot[bot] committed:
Bump pygments from 2.2.0 to 2.7.4 in /docs

Bumps [pygments](https://github.com/pygments/pygments) from 2.2.0 to 2.7.4.
- [Release notes](https://github.com/pygments/pygments/releases)
- [Changelog](https://github.com/pygments/pygments/blob/master/CHANGES)
- [Commits](https://github.com/pygments/pygments/compare/2.2.0...2.7.4)

Signed-off-by: dependabot[bot] <support@github.com>

chick committed:
build riscv-tools 1

jerryz123 committed:
Bump testchipip | Add custom boot pin

mysoreanoop committed:
Update barstools.rst

Added info about adding our own sram-macros cache

jerryz123 committed:
Bump testchipip

mysoreanoop committed:
Updated barstools.rst after review-1

Edits after review-1

chick committed:
build riscv-tools 1

chick committed:
build riscv-tools 1

abejgonzalez commented on .github/wip/create-hashes.yml:
```diff
@@ -0,0 +1,44 @@
+    name: create-hashes
+    runs-on: ubuntu-latest
+    container:
+      image: ucbbar/chipyard-image:1.0.1
```
```suggestion
      image: ucbbar/chipyard-ci-image:9c650dea
```

chick committed:
change docker container

chick committed:
change docker container

chick committed:
change docker container

abejgonzalez commented on .github/scripts/defaults.sh:
```diff
@@ -0,0 +1,104 @@
+# verilator version
+VERILATOR_VERSION=v4.034
+
+pwd
```
```suggestion
pwd
git --version
```

abejgonzalez committed:
Update .github/scripts/defaults.sh

abejgonzalez commented on .github/workflows/chipyard-ci-process.yml:
```diff
@@ -0,0 +1,238 @@
+    env:
+      JVM_OPTS: -Xmx3200m # Customize the JVM maximum heap limit
+    steps:
+#      - name: fix permissions 1
+#        run: sudo chown -R riscvuser .
```
```suggestion
      - name: Check git version
        run: git --version
```

abejgonzalez committed:
Update .github/workflows/chipyard-ci-process.yml

abejgonzalez committed:
Bump requirements to get newer Git version (for CI)

abejgonzalez committed:
Enable add-apt-repo

jerryz123 committed:
Merge pull request #843 from ucb-bar/chip-features

Features for test chips

abejgonzalez committed:
Bump CI image

abejgonzalez committed:
Use new Docker image with updated Git (2.31.X)

chick committed:
change docker container

jerryz123 committed:
Bump testchipip

chick committed:
change docker container

abejgonzalez committed:
Support 30MiB payloads - VCU118 FPGA

abejgonzalez committed:
Use multi-clock config. frags to determine VCU118 clk freq

chick committed:
try to show df as build riscv runs

abejgonzalez committed:
Bump FireMarshal for prototype targets

abejgonzalez committed:
Set both MBUS/PBUS in configs | Add simple check for correct clocks

abejgonzalez committed:
Small comment updates + cleanup

abejgonzalez committed:
First commit of docs [ci skip]

abejgonzalez committed:
Cleanup docs for Linux on VCU118

abejgonzalez committed:
Merge pull request #846 from ucb-bar/update-git-docker

Update `git` in Docker image

abejgonzalez committed:
Merge pull request #838 from ucb-bar/dependabot/pip/docs/pygments-2.7.4

Bump pygments from 2.2.0 to 2.7.4 in /docs

mysoreanoop committed:
Merge branch 'dev' into patch-2

mysoreanoop committed:
Update Barstools.rst

mysoreanoop committed:
conflict resolution

colinschmidt committed:
Merge pull request #845 from mysoreanoop/patch-2

Update barstools.rst

abejgonzalez committed:
Bump FireMarshal

colinschmidt committed:
Workaround docutils issue in doc generation

Need this until the following is fixed: https://github.com/sphinx-doc/sphinx/issues/9063

abejgonzalez committed:
Merge pull request #851 from ucb-bar/doc-fix

Workaround docutils issue in doc generation

abejgonzalez committed:
Merge remote-tracking branch 'origin/dev' into vcu118-update

abejgonzalez committed:
Bump FireMarshal

Commit:
Bump the spec workload to get the marshal overhaul

abejgonzalez committed:
Merge pull request #852 from ucb-bar/bumpSpec

Bump the spec workload to get the marshal overhaul

alfonrod committed:
Ensure proper alignment with 'CacheBlockBytes' in charcount test

jerryz123 committed:
Merge pull request #853 from alfonrod/fix-tests-charcount

Fix memory alignment in character count RoCC test

jerryz123 committed:
Merge pull request #848 from ucb-bar/chip-features

Bump testchipip for new TLROM over Serial

abejgonzalez committed:
Update grammer issues a bit

chick committed:
try to show df as build riscv runs

chick committed:
see if cache works with riscv-tools

chick committed:
see if cache works with riscv-tools

abejgonzalez committed:
Address review comments

abejgonzalez committed:
Update ref in chip communication docs.

abejgonzalez committed:
Apply suggestions from code review

Co-authored-by: alonamid <alonamid@eecs.berkeley.edu>

abejgonzalez committed:
Update VCU118.rst

abejgonzalez committed:
Merge pull request #849 from ucb-bar/vcu118-update

VCU118 FPGA Updates + FireMarshal on Prototypes

chick committed:
cleaned up process
added check commits

chick committed:
add tutorial setup job

chick committed:
add documentation test

chick committed:
debugging documentation error

chick committed:
debugging documentation error

chick committed:
debugging documentation error

chick committed:
Merge remote-tracking branch 'upstream/dev' into github-actions-port-2

chick committed:
debugging documentation error

chick committed:
add build esp toolchain build

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
testing ssh to millennium
building verilator there

chick committed:
working on build extra tests

chick committed:
working on build extra tests
combining with previous steps

chick committed:
working on build extra tests
combining with previous steps

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
some cleanup work

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick committed:
trying to do better keys cache

chick commented:
working on next pr now. closing this
